### PR TITLE
Make CatalogSource visible in the API

### PR DESCRIPTION
**Pull request by AlpAribal**
# Description

Adds `CatalogSource` to the `__all__` list to make it visible in our API.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [x] Lint and unit tests pass locally with my changes
- [x] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: squirrel/catalog/catalog.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
     from squirrel.driver import Driver
 
-__all__ = ["Catalog", "CatalogKey"]
+__all__ = ["Catalog", "CatalogKey", "CatalogSource"]
```
[the remark this answers is not in the document]
Comment (mg515) [reply squirrel/catalog/catalog.py:32]: I'm not sure if this counts as using the `public api`, but here is an [example](https://github.com/merantix-momentum/chameleon/blob/a525524653bf7a0a04fcfe03f844792d84b11e86/src/chameleon/data_accessor/dataset/dataset.py#L322) from chameleon, that I remembered of.

[the remark this answers is not in the document]
Comment (AlpAribal) [reply squirrel/catalog/catalog.py:32]: Here are my reasons:
- `CatalogSource` is not a class that is only being used internally, any user that has a Catalog ends up with a `CatalogSource` and also calls `CatalogSource.get_driver()`.
- `CatalogSource` has more to offer than `Source` (e.g. `get_driver()`, `identifier`, etc.). Without making the class visible, we cannot document them, or the other way around, if we want to document them why make the class invisible?

[the remark this answers is not in the document]
Comment (AlirezaSohofi) [reply squirrel/catalog/catalog.py:32]: Makes sense. I think then `CatalogSource` should be mentioned in the doc with some examples maybe.

**Review by AlirezaSohofi — APPROVED** [review]
:+1:

**Pull request merged**